agent-control/src/packages/oci/downloader.rs

```diff
@@ -0,0 +1,321 @@
+use crate::http::config::ProxyConfig;
+use crate::utils::certificates::{cert_paths_from_dir, certificate_error};
+use oci_client::client::{Certificate, CertificateEncoding, ClientConfig};
+use oci_client::{Client, secrets::RegistryAuth};
+use oci_spec::distribution::Reference;
+use rustls_pki_types::pem::PemObject;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use tokio;
+use tracing::debug;
+use url::Url;
+
+#[derive(Debug, Error)]
+pub enum OCIDownloaderError {
+    #[error("Donwnloading OCI Artifact: {0}")]
+    DownloadingArtifactError(String),
+    #[error("certificate error: {0}")]
+    CertificateError(String),
+}
+
+#[allow(dead_code, reason = "still unused")]
```

Comment:
We'll need this struct to be public and maybe the dead_code warning already goes away,